We link to it unconditionally, as otherwise we get link failures if
some objects are -fstack-protector built and final link happens with
-fno-stack-protector. This seems to be the common case when
bootstrapping gcc, the piepatches do not seem to fully fix the
crosstoolchain and  bootstrap sequence wrt. stack-protector flag usage.

Based on patch by Timo Teras for Alpine Linux.

--- gcc-8-20180907/gcc/gcc.c.orig	2018-02-09 03:44:06.000000000 -0300
+++ gcc-8-20180907/gcc/gcc.c	2018-09-12 17:43:14.078635112 -0300
@@ -869,8 +869,7 @@
 
 #ifndef LINK_SSP_SPEC
 #ifdef TARGET_LIBC_PROVIDES_SSP
-#define LINK_SSP_SPEC "%{fstack-protector|fstack-protector-all" \
-		       "|fstack-protector-strong|fstack-protector-explicit:}"
+#define LINK_SSP_SPEC "-lssp_nonshared"
 #else
 #define LINK_SSP_SPEC "%{fstack-protector|fstack-protector-all" \
 		       "|fstack-protector-strong|fstack-protector-explicit" \
